### ISSUE TITLE
Delete dynamic_propagation config

### ISSUE THREAD

**Issue by ezyang** [pytorch/torchdynamo#1949]

**Comment by eellison** [comment]
I could see this being possibly being useful for export in some contexts, or for model manipulation. For example, if you had a Conv-Batchnorm block, you could first compile with dynamic propagation, do model manipulation, and then use that cached manipulation for subsequent uses on different shapes. 

But the details there are hazy and it might not work so maybe we should delete this.

**Comment by ezyang** [comment]
to be clear, dynamic propagation defaults to true, and removing it would remove the ability to set it to false

**Comment by jansel** [comment]
This makes sense.  The point of the flag here was to save memory, which should not be an issue with fake tensors.
